data-products/src/main/scala/org/sunbird/analytics/sourcing/SourcingSummaryReport.scala

```diff
@@ -0,0 +1,129 @@
+    val projectDf = programData.join(nominationData, Seq("program_id"),"outer")
+      .select("program_id","status","rootorg_id","user_id")
+    JobLogger.log(s"Project count - ${projectDf.count()}",None, Level.INFO)
+
+    process(projectDf)
+  }
+
+  def process(projectDf: DataFrame)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig) = {
+    val userDf = getUserDetails()
+    JobLogger.log(s"User data count - ${userDf.count()}",None, Level.INFO)
+    val resultDf = userDf.join(projectDf, userDf.col("userId") === projectDf.col("user_id"), "outer")
+      .withColumnRenamed("user_id","contributorId")
+      .withColumn("reportName", lit("SourcingReport"))
+      .withColumn("timestamp", lit(System.currentTimeMillis()))
+    JobLogger.log(s"resultDf count - ${resultDf.count()}",None, Level.INFO)
+
+    val modelParams = config.modelParams.get
+    val storageConfig = StorageConfig(AppConf.getConfig("cloud_storage_type"), "reports", "", Option(modelParams.getOrElse("storageKey","druid_storage_account_key").toString), Option(modelParams.getOrElse("storageSecret","druid_storage_account_secret").toString))
+    resultDf.saveToBlobStore(storageConfig, "json", "sourcing",
+      Option(Map("header" -> "true")), Option(List("reportName")))
+    submitReportToDruid(modelParams)
+  }
+
+  def submitReportToDruid(modelParams: Map[String, AnyRef]): Unit = {
+    JobLogger.log(s"Submitting Druid Ingestion Task", None, Level.INFO)
+    val segmentUrl = modelParams.getOrElse("druidSegmentUrl", "http://localhost:8081/druid/coordinator/v1/metadata/datasources/sourcing-model-snapshot/segments").asInstanceOf[String]
+    val deleteSegmentUrl = modelParams.getOrElse("deleteSegmentUrl", "http://localhost:8081/druid/coordinator/v1/datasources/sourcing-model-snapshot/segments/").asInstanceOf[String]
+    val olderSegments = RestUtil.get[List[String]](segmentUrl)
+    JobLogger.log(s"olderSegments - $olderSegments",None, Level.INFO)
+    //disable older segments
+    if(null != olderSegments) {
+      olderSegments.foreach(segmentId => {
+        val apiUrl = deleteSegmentUrl+segmentId
+        RestUtil.delete(apiUrl)
+        JobLogger.log(s"Deleted $segmentId",None, Level.INFO)
+      })
+    }
+
+    val ingestionSpecPath: String = modelParams.getOrElse("specPath", "").asInstanceOf[String]
+    val druidIngestionUrl: String = modelParams.getOrElse("druidIngestionUrl", "http://localhost:8081/druid/indexer/v1/task").asInstanceOf[String]
+    CollectionSummaryJobV2.submitIngestionTask(druidIngestionUrl, ingestionSpecPath)
+    JobLogger.log(s"Druid ingestion completed", None, Level.INFO)
+  }
+
+  def getUserDetails()(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): DataFrame = {
+    val openSaberDb = config.modelParams.get.getOrElse("dbName", "opensaberdb")
+    val dbUrl = AppConf.getConfig("postgres.url") + openSaberDb
+
+    val vUserData = spark.read.jdbc(dbUrl, tables("vUser"), connProperties)
```

Comment:
It would be good to keep the data loading logic in one place and invoke that method.  instead of writing the same core code multiple places 
`spark.read.jdbc()`